### PR TITLE
Smaller fixes for Angular's new compiler

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: src/tsickle.ts [hunk 1]
```diff
@@ -2129,3 +2125,7 @@ function combineSourceMaps(
   }
   return tscSourceMapGenerator ? tscSourceMapGenerator.toString() : tscSourceMapText;
 }
+
+function normalizeArray<T>(arr: T[]|undefined): T[] {
+  return arr ? arr : [];
+}
```
Comment (evmar) [inline src/tsickle.ts]: Maybe include trailing newline here?

File: src/tsickle.ts [hunk 2]
```diff
@@ -2129,3 +2125,7 @@ function combineSourceMaps(
   }
   return tscSourceMapGenerator ? tscSourceMapGenerator.toString() : tscSourceMapText;
 }
+
+function normalizeArray<T>(arr: T[]|undefined): T[] {
```
Comment (evmar) [inline src/tsickle.ts]: Is this the same as writing
 [
    ...(foo || []),
   ...(bar || []),
]

above?  It might make more sense to inline, I dunno.

Comment (tbosch) [reply src/tsickle.ts]: Yes, good point.

File: src/tsickle.ts (hunk 1 above)
Comment (tbosch) [reply src/tsickle.ts]: Removed this function, see above.

**Pull request merged**